src/mxarray.jl

```diff
@@ -655,10 +655,10 @@ function jstring(mx::MxArray)
         throw(ArgumentError("jstring only applies to strings (i.e. char vectors)."))
     end
     len = ncols(mx) + 2
-    tmp = Array(UInt8, len)
+    tmp = Array(UInt8, len-2)
     ccall(_mx_get_string, Cint, (Ptr{Void}, Ptr{UInt8}, mwSize), 
         mx.ptr, tmp, len)
-    bytestring(pointer(tmp))
+    String(tmp)
```

Comment:
This is not right since you are telling MATLAB to write past the end of the array that was allocated. The best solution I can see is:

```julia
function jstring(mx::MxArray)
    if !(classid(mx) == mxCHAR_CLASS && ((ndims(mx) == 2 && nrows(mx) == 1) || is_empty(mx)))
        throw(ArgumentError("jstring only applies to strings (i.e. char vectors)."))
    end
    len = ncols(mx) + 1
    tmp = Array(UInt8, len)
    ccall(_mx_get_string, Cint, (Ptr{Void}, Ptr{UInt8}, mwSize), 
          mx.ptr, tmp, len)
    pop!(tmp)
    String(tmp)
end
```